src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -873,42 +873,24 @@ public void ValidateRelationshipsInConfig(RuntimeConfig runtimeConfig, IMetadata
                     }
                 }
 
-                // In all kinds of relationships, if sourceFields are included they must be valid columns in the backend.
-                if (relationship.SourceFields is not null)
+                try
                 {
-                    GetFieldsNotBackedByColumnsInDB(
-                        fields: relationship.SourceFields,
+                    ValidateSourceAndTargetFieldsAsBackingColumns(
                         invalidColumns: invalidColumns,
+                        sourceFields: relationship.SourceFields,
+                        targetFields: relationship.TargetFields,
+                        relationshipName: relationshipName,
                         entityName: entityName,
+                        targetEntityName: relationship.TargetEntity,
                         sqlMetadataProvider: sqlMetadataProvider);
-
-                    if (invalidColumns.Count > 0)
-                    {
-                        HandleOrRecordException(new DataApiBuilderException(
-                            message: $"Entity: {entityName} has a relationship: {relationshipName} with source fields: {string.Join(",", invalidColumns)} that " +
-                                $"do not exist as columns in entity: {entityName}.",
-                            statusCode: HttpStatusCode.ServiceUnavailable,
-                            subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError));
-                    }
                 }
-
-                // In all kinds of relationships, if targetFields are included they must be valid columns in the backend.
-                if (relationship.TargetFields is not null)
+                catch (KeyNotFoundException ex)
                 {
-                    GetFieldsNotBackedByColumnsInDB(
-                        fields: relationship.TargetFields,
-                        invalidColumns: invalidColumns,
-                        entityName: relationship.TargetEntity,
-                        sqlMetadataProvider: sqlMetadataProvider);
-
-                    if (invalidColumns.Count > 0)
-                    {
-                        HandleOrRecordException(new DataApiBuilderException(
-                            message: $"Entity: {entityName} has a relationship: {relationshipName} with target fields: {string.Join(",", invalidColumns)} that " +
-                                $"do not exist as columns in entity: {relationship.TargetEntity}.",
-                            statusCode: HttpStatusCode.ServiceUnavailable,
-                            subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError));
-                    }
+                    HandleOrRecordException(new DataApiBuilderException(
+                        message: $"Unable to validate source and target fields as backing columns in the DB due to: {ex.Message} \n" +
```

Comment:
We talked about this offline, but just to give context here, we will use a custom message for the key not found so that it is useful to the end user, but remove the connection string messaging because it is redundant to other exceptions that will already communicate this information in our exception/error handling.